File: ios/Plugin/YemCapacitorFacebookPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(YemCapacitorFacebookPlugin)
public class YemCapacitorFacebookPlugin: CAPPlugin {
    private let implementation = YemCapacitorFacebook()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}

File: android/src/main/java/com/yem/plugin/capacitorfacebook/YemCapacitorFacebookPlugin.java
package com.yem.plugin.capacitorfacebook;

import com.facebook.FacebookSdk;
// import com.facebook.ads.AdView;
// import com.facebook.ads.AudienceNetworkAds;
import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "YemCapacitorFacebook")
public class YemCapacitorFacebookPlugin extends Plugin {

    private YemCapacitorFacebook implementation = new YemCapacitorFacebook();

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void initialize(PluginCall call) {
        // Retrieve the Facebook App ID from the plugin call
        String facebookAppId = call.getString("facebookAppId");

        // Initialize Facebook SDK with the provided App ID
        FacebookSdk.sdkInitialize(getContext());

        // You can perform other initialization tasks here

        call.resolve();
    }
}
